### PR TITLE
Support read replicas

### PR DESCRIPTION
**Pull request by fernandezcuesta**
- Support read replicas by deploying a separate statefulset of N-1 replicas
- Add additional ingresses to the read replica services
- Optionally install grafana dashboard via Grafana Operator CRD

**Test Plan:**

```
k3d cluster create -a3
helm upgrade --install immudb . --set replicaCount=3 --set ingress.enabled=true
kubectl port-forward svc/immudb-rr-grpc 2233:3322 &
kubectl port-forward svc/immudb-grpc 3322:3322 &

# This should succeed
immuclient --username immudb --password immudb exec "UPSERT INTO orders (id, customerid, productid) VALUES ( 2, 1, 1);"
# This should fail
immuclient --immudb-port 2233 --username immudb --password immudb exec "UPSERT INTO orders (id, customerid, productid) VALUES ( 2, 1, 1);"
```

### CODE REVIEW

File: helm/templates/dashboard.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.grafanaDashboard -}}
```
Comment (byo) [inline helm/templates/dashboard.yaml:1]: Let's make this `.Values.grafanaDashboard.enabled` - that way the `.Values.grafanaDashboard` object can be easily extended with other options in backwards-compatible way.

File: helm/templates/dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDashboard
+metadata: 
+  name: immudb-grafana-dashboard
+  labels: 
+    app: immudb
+spec:
+  url: https://raw.githubusercontent.com/codenotary/immudb/master/tools/monitoring/grafana-dashboard.json
```
Comment (byo) [inline helm/templates/dashboard.yaml:9]: We could use URL for specific immudb version here, e.g. `https://raw.githubusercontent.com/codenotary/immudb/v1.3.2/tools/monitoring/grafana-dashboard.json`

that could default to the same version as the immudb one that is used in the statefulset with an option to override, e.g.:

```suggestion
  url: https://raw.githubusercontent.com/codenotary/immudb/{{- .Values.grafanaDashboard.version | default print "v" .Chart.AppVersion -}}/tools/monitoring/grafana-dashboard.json
```

File: helm/templates/statefulset.yaml [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
     spec:
-      {{- with .Values.imagePullSecrets }}
+      {{- with $.Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      securityContext:
-        {{- toYaml .Values.podSecurityContext | nindent 8 }}
-      volumes:
-      - name: immudb-storage
```
Comment (byo) [inline helm/templates/statefulset.yaml:32]: I see the volume removed but where it has been moved to? I can't find the place.

File: helm/templates/ingress.yaml [hunk 4]
```diff
@@ -1,52 +1,61 @@
...
-  name: {{ $fullName }}-http
+  name: {{ . }}-http
   labels:
-    {{- include "immudb.labels" . | nindent 4 }}
-  {{- with .Values.ingress.annotations }}
+    {{- include "immudb.labels" $ | nindent 4 }}
+  {{- with $.Values.ingress.annotations }}
   annotations:
-    {{- toYaml . | nindent 4 }}
+    {{- toYaml $ | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (fernandezcuesta) [reply helm/templates/ingress.yaml:25]: `.` here refers to the `range` loop, whereas `$` refers to the context out from the loop

File: helm/templates/statefulset.yaml (hunk 3 above)
Comment (fernandezcuesta) [reply helm/templates/statefulset.yaml:32]: If I'm not mistaken, that's not needed (see example [here](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#components))

Comment (SimoneLazzaris) [reply helm/templates/statefulset.yaml:32]: It is not needed *if* you provide a volumeClaimTemplate. You must provide one when using more than one replicas

**Comment by jeroiraz** [comment]
@SimoneLazzaris could you review this changes?

**Pull request closed without merging**